Update `aggregator` doc (#1365)
scroll-tech/zkevm-circuits: aggregator/README.md

## Batch

A __batch__ is a list of continuous `chunks` of size `k` that will be aggregated using the `BatchCircuit`. If the input chunks' size `k` is less than `n`, we pad the input with `(n-k)` chunks identical to `chunk[k]`. The batch is represented by the preimage fields to the `batch_hash`, which is constructed as:
```
batchHash := keccak256(version || batch_index || l1_message_popped || total_l1_message_popped || batch_data_hash || versioned_hash || parent_batch_hash || last_block_timestamp || z || y)
```
All preimage fields' values are provided to the batch through the `BatchHeader` struct, so it can correctly construct the hash state transition from `parent_batch_hash` to `batch_hash` (for current batch). 

> Note there are also implicitly represents of state roots before/after batch from the states of chunks it has aggregated:

Author: ```suggestion
Note that there are also implicit constraints between state roots before/after batch and the state roots of the chunks it has aggregated:
```

## Bundle
A __bundle__ is a list of continuous `batches` that will be aggregated recursively using the `RecursionCircuit`. The __bundle__ is the current apex entity whose proof will be verified on-chain.

# Circuits

## Chunk circuit

Circuit proving the relationship for a chunk is the zkEVM circuit. It will go through 2 layers of compression circuit, and becomes a __snark__ struct. We do not list its details here. Abstractly, a snark circuit has the following properties:
- it takes 44 elements as public inputs 
    - 12 from accumulators
    - 32 from public input hash

## Batch Circuit

We want to aggregate `k` snarks, each from a valid chunk. We generate `(n-k)` padded chunks (by repeating the last non-padding chunk), and obtain a total of `n` snarks.
Additionally, the batch circuit has to ascertain the correct hash transition from the parent batch. 

### Configuration

There are several configuration subcomponents for batch circuit.
- FpConfig; used for snark aggregation.
- KeccakConfig: used to build keccak table.
- RlcConfig: used to compute RLC of hash inputs.

> - BlobDataConfig: used for representing the zstd-encoded form of batch data, with 4096 * 31 rows. Each row is a byte value.

Author: ```suggestion
- BlobDataConfig: used for representing the zstd-encoded form of batch data, with `4096 * 31` rows. Each row is a byte value. An EIP-4844 blob consists of `4096 * 32` bytes, where we set the most-significant byte in each 32-bytes chunk as `0` to guarantee that each 32-bytes chunk is a valid BLS12-381 scalar field element.
```

> - BatchDataConfig: used for representing plain bytes as preimage for `batch_hash`.

Author: ```suggestion
- BatchDataConfig: used for representing the raw batch bytes, effectively constructing the random challenge point `z` for the KZG opening proof.
```

> - DecoderConfig: used for describing how the blob data is decoded into plain bytes (batch data).

Author: ```suggestion
- DecoderConfig: implements an in-circuit zstd-decoder that decodes blob data into batch data
```

> - BarycentricEvaluationConfig: used for evaluate the interpolated blob polynomial at an arbitrary challenge point.

Author: ```suggestion
- BarycentricEvaluationConfig: used for evaluating the interpolated blob polynomial at an arbitrary challenge point `z`, where both `z` and the evaluation `y` are included in the `BatchHeader`.
```

### Statements
For snarks $s_1,\dots,s_k,\dots, s_n$ the batch circuit argues the following statements.

1. batch_data_hash digest is reused for batch hash. __Static__.

2. batch_data_hash and chunk[i].pi_hash use a same chunk[i].data_hash when chunk[i] is not padded
```
for i in 1 ... n
    chunk_pi_hash   := keccak(chain_id || prev_state_root || post_state_root || withdraw_root || chunk_data_hash || chunk_txdata_hash)
```
This is done by computing the RLCs of chunk[i]'s data_hash for `i=0..k`, and then check the RLC matches the one from the keccak table.

3. chunks are continuous when they are not padded: they are linked via the state roots.
```
for i in 1 ... k-1
    c_i.post_state_root == c_{i+1}.prev_state_root
```
4. All the chunks use the same chain id. __Static__.
```
for i in 1 ... n
    batch.chain_id == chunk[i].chain_id
```

> 5. The last `(n-k)` chunk[i] are padding

Author: ```suggestion
5. The last `(n-k)` chunk[i] are padded chunks
```
